CI hangs in MongodbPanacheMockingTest

https://github.com/quarkusio/quarkus/pull/14385/checks?check_run_id=1732090582 hung at this point:

```
2021-01-20T04:39:05.5857670Z [INFO] Running io.quarkus.it.mongodb.panache.MongodbPanacheMockingTest
2021-01-20T05:37:40.3090133Z 2021-01-20 05:37:40
2021-01-20T05:37:40.3096977Z Full thread dump OpenJDK 64-Bit Server VM (11.0.9.1+1 mixed mode):
2021-01-20T05:37:40.3098538Z 
2021-01-20T05:37:40.3099435Z Threads class SMR info:
2021-01-20T05:37:40.3100186Z _java_thread_list=0x000001ca66399140, length=23, elements={
2021-01-20T05:37:40.3107212Z 0x000001ca2ab24000, 0x000001ca47674800, 0x000001ca4767d800, 0x000001ca47fe8800,
2021-01-20T05:37:40.3114501Z 0x000001ca47fea800, 0x000001ca47fec000, 0x000001ca47f94000, 0x000001ca47f9a000,
2021-01-20T05:37:40.3115737Z 0x000001ca48153800, 0x000001ca4815e000, 0x000001ca49101000, 0x000001ca48f7f800,
2021-01-20T05:37:40.3116744Z 0x000001ca491fd800, 0x000001ca49190800, 0x000001ca485fd000, 0x000001ca50066800,
2021-01-20T05:37:40.3119727Z 0x000001ca771dd800, 0x000001ca771e4000, 0x000001ca771e5000, 0x000001ca771e7800,
2021-01-20T05:37:40.3120731Z 0x000001ca771e6000, 0x000001ca771e6800, 0x000001ca771e2800
2021-01-20T05:37:40.3121374Z }
2021-01-20T05:37:40.3121720Z 
2021-01-20T05:37:40.3124745Z "main" #1 prio=5 os_prio=0 cpu=721171.88ms elapsed=10635.89s tid=0x000001ca2ab24000 nid=0x1740 runnable  [0x00000029fe4fd000]
2021-01-20T05:37:40.3125693Z    java.lang.Thread.State: RUNNABLE
2021-01-20T05:37:40.3127015Z 	at java.lang.ProcessImpl.waitForInterruptibly(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3129185Z 	at java.lang.ProcessImpl.waitFor(java.base@11.0.9.1/ProcessImpl.java:545)
2021-01-20T05:37:40.3131435Z 	at org.apache.maven.surefire.extensions.util.CommandlineExecutor.awaitExit(CommandlineExecutor.java:90)
2021-01-20T05:37:40.3134304Z 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:654)
2021-01-20T05:37:40.3136867Z 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:305)
2021-01-20T05:37:40.3139256Z 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:265)
2021-01-20T05:37:40.3142184Z 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1314)
2021-01-20T05:37:40.3145922Z 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1159)
2021-01-20T05:37:40.3149455Z 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:932)
2021-01-20T05:37:40.3152226Z 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
2021-01-20T05:37:40.3155618Z 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
2021-01-20T05:37:40.3157875Z 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
2021-01-20T05:37:40.3160084Z 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
2021-01-20T05:37:40.3162919Z 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
2021-01-20T05:37:40.3166324Z 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
2021-01-20T05:37:40.3171148Z 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
2021-01-20T05:37:40.3175654Z 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
2021-01-20T05:37:40.3177675Z 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
2021-01-20T05:37:40.3179094Z 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
2021-01-20T05:37:40.3180474Z 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
2021-01-20T05:37:40.3181794Z 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
2021-01-20T05:37:40.3182998Z 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
2021-01-20T05:37:40.3184206Z 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
2021-01-20T05:37:40.3185822Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3188164Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(java.base@11.0.9.1/NativeMethodAccessorImpl.java:62)
2021-01-20T05:37:40.3190771Z 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@11.0.9.1/DelegatingMethodAccessorImpl.java:43)
2021-01-20T05:37:40.3192791Z 	at java.lang.reflect.Method.invoke(java.base@11.0.9.1/Method.java:566)
2021-01-20T05:37:40.3194584Z 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
2021-01-20T05:37:40.3197671Z 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
2021-01-20T05:37:40.3200029Z 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
2021-01-20T05:37:40.3204022Z 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
2021-01-20T05:37:40.3205565Z 
2021-01-20T05:37:40.3206406Z "Reference Handler" #2 daemon prio=10 os_prio=2 cpu=62.50ms elapsed=10635.85s tid=0x000001ca47674800 nid=0x1a64 waiting on condition  [0x00000029febff000]
2021-01-20T05:37:40.3207448Z    java.lang.Thread.State: RUNNABLE
2021-01-20T05:37:40.3212817Z 	at java.lang.ref.Reference.waitForReferencePendingList(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3216940Z 	at java.lang.ref.Reference.processPendingReferences(java.base@11.0.9.1/Reference.java:241)
2021-01-20T05:37:40.3219229Z 	at java.lang.ref.Reference$ReferenceHandler.run(java.base@11.0.9.1/Reference.java:213)
2021-01-20T05:37:40.3220060Z 
2021-01-20T05:37:40.3220853Z "Finalizer" #3 daemon prio=8 os_prio=1 cpu=21875.00ms elapsed=10635.85s tid=0x000001ca4767d800 nid=0x744 in Object.wait()  [0x00000029fecfe000]
2021-01-20T05:37:40.3223173Z    java.lang.Thread.State: WAITING (on object monitor)
2021-01-20T05:37:40.3250391Z 	at java.lang.Object.wait(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3251433Z 	- waiting on <no object reference available>
2021-01-20T05:37:40.3252594Z 	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.9.1/ReferenceQueue.java:155)
2021-01-20T05:37:40.3283067Z 	- waiting to re-lock in wait() <0x00000000a81f2728> (a java.lang.ref.ReferenceQueue$Lock)
2021-01-20T05:37:40.3284617Z 	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.9.1/ReferenceQueue.java:176)
2021-01-20T05:37:40.3286142Z 	at java.lang.ref.Finalizer$FinalizerThread.run(java.base@11.0.9.1/Finalizer.java:170)
2021-01-20T05:37:40.3286849Z 
2021-01-20T05:37:40.3287831Z "Signal Dispatcher" #4 daemon prio=9 os_prio=2 cpu=0.00ms elapsed=10635.84s tid=0x000001ca47fe8800 nid=0x19ac waiting on condition  [0x0000000000000000]
2021-01-20T05:37:40.3288831Z    java.lang.Thread.State: RUNNABLE
2021-01-20T05:37:40.3290740Z 
2021-01-20T05:37:40.3291601Z "Attach Listener" #5 daemon prio=5 os_prio=2 cpu=0.00ms elapsed=10635.84s tid=0x000001ca47fea800 nid=0x8b4 runnable  [0x0000000000000000]
2021-01-20T05:37:40.3292513Z    java.lang.Thread.State: RUNNABLE
2021-01-20T05:37:40.3292967Z 
2021-01-20T05:37:40.3293847Z "C2 CompilerThread0" #6 daemon prio=9 os_prio=2 cpu=568859.38ms elapsed=10635.84s tid=0x000001ca47fec000 nid=0x19f0 waiting on condition  [0x0000000000000000]
2021-01-20T05:37:40.3294923Z    java.lang.Thread.State: RUNNABLE
2021-01-20T05:37:40.3297008Z    No compile task
2021-01-20T05:37:40.3297367Z 
2021-01-20T05:37:40.3298163Z "C1 CompilerThread0" #7 daemon prio=9 os_prio=2 cpu=48968.75ms elapsed=10635.84s tid=0x000001ca47f94000 nid=0xc78 waiting on condition  [0x0000000000000000]
2021-01-20T05:37:40.3299192Z    java.lang.Thread.State: RUNNABLE
2021-01-20T05:37:40.3299893Z    No compile task
2021-01-20T05:37:40.3301799Z 
2021-01-20T05:37:40.3302640Z "Sweeper thread" #8 daemon prio=9 os_prio=2 cpu=14156.25ms elapsed=10635.84s tid=0x000001ca47f9a000 nid=0x74c runnable  [0x0000000000000000]
2021-01-20T05:37:40.3303551Z    java.lang.Thread.State: RUNNABLE
2021-01-20T05:37:40.3304092Z 
2021-01-20T05:37:40.3304849Z "Service Thread" #9 daemon prio=9 os_prio=0 cpu=0.00ms elapsed=10635.79s tid=0x000001ca48153800 nid=0x51c runnable  [0x0000000000000000]
2021-01-20T05:37:40.3307197Z    java.lang.Thread.State: RUNNABLE
2021-01-20T05:37:40.3307837Z 
2021-01-20T05:37:40.3309279Z "Common-Cleaner" #10 daemon prio=8 os_prio=1 cpu=140.63ms elapsed=10635.78s tid=0x000001ca4815e000 nid=0x694 in Object.wait()  [0x00000029ff4fe000]
2021-01-20T05:37:40.3310461Z    java.lang.Thread.State: TIMED_WAITING (on object monitor)
2021-01-20T05:37:40.3313125Z 	at java.lang.Object.wait(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3314123Z 	- waiting on <no object reference available>
2021-01-20T05:37:40.3315820Z 	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.9.1/ReferenceQueue.java:155)
2021-01-20T05:37:40.3318866Z 	- waiting to re-lock in wait() <0x00000000a8113288> (a java.lang.ref.ReferenceQueue$Lock)
2021-01-20T05:37:40.3320301Z 	at jdk.internal.ref.CleanerImpl.run(java.base@11.0.9.1/CleanerImpl.java:148)
2021-01-20T05:37:40.3321453Z 	at java.lang.Thread.run(java.base@11.0.9.1/Thread.java:834)
2021-01-20T05:37:40.3324350Z 	at jdk.internal.misc.InnocuousThread.run(java.base@11.0.9.1/InnocuousThread.java:134)
2021-01-20T05:37:40.3325203Z 
2021-01-20T05:37:40.3326049Z "resolver-1" #36 daemon prio=5 os_prio=0 cpu=421.88ms elapsed=10600.13s tid=0x000001ca49101000 nid=0x7b0 waiting on condition  [0x00000029ff7ff000]
2021-01-20T05:37:40.3327128Z    java.lang.Thread.State: WAITING (parking)
2021-01-20T05:37:40.3329509Z 	at jdk.internal.misc.Unsafe.park(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3331200Z 	- parking to wait for  <0x00000000a86c2e58> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
2021-01-20T05:37:40.3334087Z 	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.9.1/LockSupport.java:194)
2021-01-20T05:37:40.3336559Z 	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.9.1/AbstractQueuedSynchronizer.java:2081)
2021-01-20T05:37:40.3338969Z 	at java.util.concurrent.LinkedBlockingQueue.take(java.base@11.0.9.1/LinkedBlockingQueue.java:433)
2021-01-20T05:37:40.3342262Z 	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.9.1/ThreadPoolExecutor.java:1054)
2021-01-20T05:37:40.3344662Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.9.1/ThreadPoolExecutor.java:1114)
2021-01-20T05:37:40.3347003Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.9.1/ThreadPoolExecutor.java:628)
2021-01-20T05:37:40.3349345Z 	at java.lang.Thread.run(java.base@11.0.9.1/Thread.java:834)
2021-01-20T05:37:40.3349819Z 
2021-01-20T05:37:40.3351137Z "resolver-2" #37 daemon prio=5 os_prio=0 cpu=312.50ms elapsed=10600.13s tid=0x000001ca48f7f800 nid=0xa24 waiting on condition  [0x00000029ff8fe000]
2021-01-20T05:37:40.3352469Z    java.lang.Thread.State: WAITING (parking)
2021-01-20T05:37:40.3355154Z 	at jdk.internal.misc.Unsafe.park(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3356793Z 	- parking to wait for  <0x00000000a86c2e58> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
2021-01-20T05:37:40.3358871Z 	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.9.1/LockSupport.java:194)
2021-01-20T05:37:40.3362312Z 	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.9.1/AbstractQueuedSynchronizer.java:2081)
2021-01-20T05:37:40.3365282Z 	at java.util.concurrent.LinkedBlockingQueue.take(java.base@11.0.9.1/LinkedBlockingQueue.java:433)
2021-01-20T05:37:40.3369356Z 	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.9.1/ThreadPoolExecutor.java:1054)
2021-01-20T05:37:40.3372349Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.9.1/ThreadPoolExecutor.java:1114)
2021-01-20T05:37:40.3375926Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.9.1/ThreadPoolExecutor.java:628)
2021-01-20T05:37:40.3377263Z 	at java.lang.Thread.run(java.base@11.0.9.1/Thread.java:834)
2021-01-20T05:37:40.3378105Z 
2021-01-20T05:37:40.3378971Z "resolver-3" #38 daemon prio=5 os_prio=0 cpu=296.88ms elapsed=10600.14s tid=0x000001ca491fd800 nid=0x314 waiting on condition  [0x00000029ff9fe000]
2021-01-20T05:37:40.3380306Z    java.lang.Thread.State: WAITING (parking)
2021-01-20T05:37:40.3382569Z 	at jdk.internal.misc.Unsafe.park(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3384375Z 	- parking to wait for  <0x00000000a86c2e58> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
2021-01-20T05:37:40.3392496Z 	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.9.1/LockSupport.java:194)
2021-01-20T05:37:40.3395013Z 	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.9.1/AbstractQueuedSynchronizer.java:2081)
2021-01-20T05:37:40.3397438Z 	at java.util.concurrent.LinkedBlockingQueue.take(java.base@11.0.9.1/LinkedBlockingQueue.java:433)
2021-01-20T05:37:40.3399509Z 	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.9.1/ThreadPoolExecutor.java:1054)
2021-01-20T05:37:40.3402389Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.9.1/ThreadPoolExecutor.java:1114)
2021-01-20T05:37:40.3405853Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.9.1/ThreadPoolExecutor.java:628)
2021-01-20T05:37:40.3407367Z 	at java.lang.Thread.run(java.base@11.0.9.1/Thread.java:834)
2021-01-20T05:37:40.3407809Z 
2021-01-20T05:37:40.3408794Z "resolver-4" #39 daemon prio=5 os_prio=0 cpu=609.38ms elapsed=10600.14s tid=0x000001ca49190800 nid=0x1294 waiting on condition  [0x00000029ffafe000]
2021-01-20T05:37:40.3409864Z    java.lang.Thread.State: WAITING (parking)
2021-01-20T05:37:40.3410815Z 	at jdk.internal.misc.Unsafe.park(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3412459Z 	- parking to wait for  <0x00000000a86c2e58> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
2021-01-20T05:37:40.3414397Z 	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.9.1/LockSupport.java:194)
2021-01-20T05:37:40.3416591Z 	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.9.1/AbstractQueuedSynchronizer.java:2081)
2021-01-20T05:37:40.3418985Z 	at java.util.concurrent.LinkedBlockingQueue.take(java.base@11.0.9.1/LinkedBlockingQueue.java:433)
2021-01-20T05:37:40.3420946Z 	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.9.1/ThreadPoolExecutor.java:1054)
2021-01-20T05:37:40.3423284Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.9.1/ThreadPoolExecutor.java:1114)
2021-01-20T05:37:40.3425217Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.9.1/ThreadPoolExecutor.java:628)
2021-01-20T05:37:40.3426522Z 	at java.lang.Thread.run(java.base@11.0.9.1/Thread.java:834)
2021-01-20T05:37:40.3426959Z 
2021-01-20T05:37:40.3427732Z "resolver-5" #40 daemon prio=5 os_prio=0 cpu=312.50ms elapsed=10599.31s tid=0x000001ca485fd000 nid=0x17dc waiting on condition  [0x00000029ffbff000]
2021-01-20T05:37:40.3428670Z    java.lang.Thread.State: WAITING (parking)
2021-01-20T05:37:40.3429593Z 	at jdk.internal.misc.Unsafe.park(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3431180Z 	- parking to wait for  <0x00000000a86c2e58> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
2021-01-20T05:37:40.3433148Z 	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.9.1/LockSupport.java:194)
2021-01-20T05:37:40.3435345Z 	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.9.1/AbstractQueuedSynchronizer.java:2081)
2021-01-20T05:37:40.3437751Z 	at java.util.concurrent.LinkedBlockingQueue.take(java.base@11.0.9.1/LinkedBlockingQueue.java:433)
2021-01-20T05:37:40.3439716Z 	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.9.1/ThreadPoolExecutor.java:1054)
2021-01-20T05:37:40.3441789Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.9.1/ThreadPoolExecutor.java:1114)
2021-01-20T05:37:40.3443731Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.9.1/ThreadPoolExecutor.java:628)
2021-01-20T05:37:40.3445030Z 	at java.lang.Thread.run(java.base@11.0.9.1/Thread.java:834)
2021-01-20T05:37:40.3445464Z 
2021-01-20T05:37:40.3446522Z "ForkJoinPool.commonPool-worker-3" #292 daemon prio=5 os_prio=0 cpu=0.00ms elapsed=10444.41s tid=0x000001ca50066800 nid=0xc98 waiting on condition  [0x00000029800ff000]
2021-01-20T05:37:40.3447692Z    java.lang.Thread.State: WAITING (parking)
2021-01-20T05:37:40.3448622Z 	at jdk.internal.misc.Unsafe.park(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3449874Z 	- parking to wait for  <0x00000000c156cb18> (a java.util.concurrent.ForkJoinPool)
2021-01-20T05:37:40.3451308Z 	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.9.1/LockSupport.java:194)
2021-01-20T05:37:40.3452967Z 	at java.util.concurrent.ForkJoinPool.runWorker(java.base@11.0.9.1/ForkJoinPool.java:1628)
2021-01-20T05:37:40.3454892Z 	at java.util.concurrent.ForkJoinWorkerThread.run(java.base@11.0.9.1/ForkJoinWorkerThread.java:183)
2021-01-20T05:37:40.3455936Z 
2021-01-20T05:37:40.3456966Z "mysql-cj-abandoned-connection-cleanup" #4814 daemon prio=5 os_prio=0 cpu=5375.00ms elapsed=5145.98s tid=0x000001ca771dd800 nid=0x6cc in Object.wait()  [0x00000029846fe000]
2021-01-20T05:37:40.3458294Z    java.lang.Thread.State: TIMED_WAITING (on object monitor)
2021-01-20T05:37:40.3459121Z 	at java.lang.Object.wait(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3459879Z 	- waiting on <no object reference available>
2021-01-20T05:37:40.3460907Z 	at java.lang.ref.ReferenceQueue.remove(java.base@11.0.9.1/ReferenceQueue.java:155)
2021-01-20T05:37:40.3462237Z 	- waiting to re-lock in wait() <0x00000000d7c80178> (a java.lang.ref.ReferenceQueue$Lock)
2021-01-20T05:37:40.3464096Z 	at com.mysql.cj.jdbc.AbandonedConnectionCleanupThread.run(AbandonedConnectionCleanupThread.java:91)
2021-01-20T05:37:40.3466495Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.9.1/ThreadPoolExecutor.java:1128)
2021-01-20T05:37:40.3468394Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.9.1/ThreadPoolExecutor.java:628)
2021-01-20T05:37:40.3469691Z 	at java.lang.Thread.run(java.base@11.0.9.1/Thread.java:834)
2021-01-20T05:37:40.3470123Z 
2021-01-20T05:37:40.3470952Z "timeout-check-timer" #9046 daemon prio=5 os_prio=0 cpu=46.88ms elapsed=3516.31s tid=0x000001ca771e4000 nid=0x1050 waiting on condition  [0x00000029801ff000]
2021-01-20T05:37:40.3471965Z    java.lang.Thread.State: TIMED_WAITING (parking)
2021-01-20T05:37:40.3472949Z 	at jdk.internal.misc.Unsafe.park(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3474542Z 	- parking to wait for  <0x00000000f6f566d0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
2021-01-20T05:37:40.3476586Z 	at java.util.concurrent.locks.LockSupport.parkNanos(java.base@11.0.9.1/LockSupport.java:234)
2021-01-20T05:37:40.3478969Z 	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(java.base@11.0.9.1/AbstractQueuedSynchronizer.java:2123)
2021-01-20T05:37:40.3481786Z 	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.9.1/ScheduledThreadPoolExecutor.java:1182)
2021-01-20T05:37:40.3484467Z 	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.9.1/ScheduledThreadPoolExecutor.java:899)
2021-01-20T05:37:40.3486875Z 	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.9.1/ThreadPoolExecutor.java:1054)
2021-01-20T05:37:40.3488901Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.9.1/ThreadPoolExecutor.java:1114)
2021-01-20T05:37:40.3490849Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.9.1/ThreadPoolExecutor.java:628)
2021-01-20T05:37:40.3492149Z 	at java.lang.Thread.run(java.base@11.0.9.1/Thread.java:834)
2021-01-20T05:37:40.3492622Z 
2021-01-20T05:37:40.3493364Z "ping-timer-10s" #9048 daemon prio=5 os_prio=0 cpu=0.00ms elapsed=3516.31s tid=0x000001ca771e5000 nid=0x1b94 waiting on condition  [0x00000029802fe000]
2021-01-20T05:37:40.3494370Z    java.lang.Thread.State: TIMED_WAITING (parking)
2021-01-20T05:37:40.3495273Z 	at jdk.internal.misc.Unsafe.park(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3496916Z 	- parking to wait for  <0x00000000f6f1ce80> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
2021-01-20T05:37:40.3498923Z 	at java.util.concurrent.locks.LockSupport.parkNanos(java.base@11.0.9.1/LockSupport.java:234)
2021-01-20T05:37:40.3501441Z 	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(java.base@11.0.9.1/AbstractQueuedSynchronizer.java:2123)
2021-01-20T05:37:40.3504327Z 	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.9.1/ScheduledThreadPoolExecutor.java:1182)
2021-01-20T05:37:40.3507026Z 	at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(java.base@11.0.9.1/ScheduledThreadPoolExecutor.java:899)
2021-01-20T05:37:40.3510045Z 	at java.util.concurrent.ThreadPoolExecutor.getTask(java.base@11.0.9.1/ThreadPoolExecutor.java:1054)
2021-01-20T05:37:40.3512164Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.9.1/ThreadPoolExecutor.java:1114)
2021-01-20T05:37:40.3514089Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.9.1/ThreadPoolExecutor.java:628)
2021-01-20T05:37:40.3515406Z 	at java.lang.Thread.run(java.base@11.0.9.1/Thread.java:834)
2021-01-20T05:37:40.3515840Z 
2021-01-20T05:37:40.3516716Z "ThreadedStreamConsumer" #9050 daemon prio=5 os_prio=0 cpu=0.00ms elapsed=3516.28s tid=0x000001ca771e7800 nid=0x1244 waiting on condition  [0x00000029803ff000]
2021-01-20T05:37:40.3517761Z    java.lang.Thread.State: WAITING (parking)
2021-01-20T05:37:40.3518685Z 	at jdk.internal.misc.Unsafe.park(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3520275Z 	- parking to wait for  <0x00000000f6f56940> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
2021-01-20T05:37:40.3522792Z 	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.9.1/LockSupport.java:194)
2021-01-20T05:37:40.3525028Z 	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@11.0.9.1/AbstractQueuedSynchronizer.java:2081)
2021-01-20T05:37:40.3527393Z 	at java.util.concurrent.ArrayBlockingQueue.take(java.base@11.0.9.1/ArrayBlockingQueue.java:417)
2021-01-20T05:37:40.3530149Z 	at org.apache.maven.plugin.surefire.booterclient.output.ThreadedStreamConsumer$Pumper.run(ThreadedStreamConsumer.java:86)
2021-01-20T05:37:40.3532428Z 	at java.lang.Thread.run(java.base@11.0.9.1/Thread.java:834)
2021-01-20T05:37:40.3532912Z 
2021-01-20T05:37:40.3533641Z "std-in-fork-1" #9052 daemon prio=5 os_prio=0 cpu=0.00ms elapsed=3516.28s tid=0x000001ca771e6000 nid=0x1a58 waiting on condition  [0x00000029804fe000]
2021-01-20T05:37:40.3534570Z    java.lang.Thread.State: WAITING (parking)
2021-01-20T05:37:40.3535505Z 	at jdk.internal.misc.Unsafe.park(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3536603Z 	- parking to wait for  <0x00000000f6f569b0> (a java.util.concurrent.Semaphore$NonfairSync)
2021-01-20T05:37:40.3538054Z 	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.9.1/LockSupport.java:194)
2021-01-20T05:37:40.3540795Z 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(java.base@11.0.9.1/AbstractQueuedSynchronizer.java:885)
2021-01-20T05:37:40.3544987Z 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(java.base@11.0.9.1/AbstractQueuedSynchronizer.java:1039)
2021-01-20T05:37:40.3549247Z 	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(java.base@11.0.9.1/AbstractQueuedSynchronizer.java:1345)
2021-01-20T05:37:40.3552102Z 	at java.util.concurrent.Semaphore.acquire(java.base@11.0.9.1/Semaphore.java:318)
2021-01-20T05:37:40.3555538Z 	at org.apache.maven.plugin.surefire.booterclient.lazytestprovider.TestLessInputStream.awaitNextCommand(TestLessInputStream.java:164)
2021-01-20T05:37:40.3561030Z 	at org.apache.maven.plugin.surefire.booterclient.lazytestprovider.TestLessInputStream.beforeNextCommand(TestLessInputStream.java:136)
2021-01-20T05:37:40.3566515Z 	at org.apache.maven.plugin.surefire.booterclient.lazytestprovider.DefaultCommandReader.readNextCommand(DefaultCommandReader.java:81)
2021-01-20T05:37:40.3570442Z 	at org.apache.maven.plugin.surefire.extensions.StreamFeeder.run(StreamFeeder.java:86)
2021-01-20T05:37:40.3571704Z 
2021-01-20T05:37:40.3572477Z "fork-1-event-thread" #9053 daemon prio=5 os_prio=0 cpu=0.00ms elapsed=3516.28s tid=0x000001ca771e6800 nid=0x750 runnable  [0x00000029805fe000]
2021-01-20T05:37:40.3573565Z    java.lang.Thread.State: RUNNABLE
2021-01-20T05:37:40.3574507Z 	at java.io.FileInputStream.readBytes(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3575696Z 	at java.io.FileInputStream.read(java.base@11.0.9.1/FileInputStream.java:279)
2021-01-20T05:37:40.3577153Z 	at java.io.BufferedInputStream.read1(java.base@11.0.9.1/BufferedInputStream.java:290)
2021-01-20T05:37:40.3578610Z 	at java.io.BufferedInputStream.read(java.base@11.0.9.1/BufferedInputStream.java:351)
2021-01-20T05:37:40.3579843Z 	- locked <0x00000000f6f1d378> (a java.io.BufferedInputStream)
2021-01-20T05:37:40.3581018Z 	at java.io.BufferedInputStream.fill(java.base@11.0.9.1/BufferedInputStream.java:252)
2021-01-20T05:37:40.3582542Z 	at java.io.BufferedInputStream.read1(java.base@11.0.9.1/BufferedInputStream.java:292)
2021-01-20T05:37:40.3583996Z 	at java.io.BufferedInputStream.read(java.base@11.0.9.1/BufferedInputStream.java:351)
2021-01-20T05:37:40.3585228Z 	- locked <0x00000000f6f1d350> (a java.io.BufferedInputStream)
2021-01-20T05:37:40.3586673Z 	at org.apache.maven.surefire.api.util.internal.Channels$3.readImpl(Channels.java:214)
2021-01-20T05:37:40.3590237Z 	at org.apache.maven.surefire.api.util.internal.AbstractNoninterruptibleReadableChannel.read(AbstractNoninterruptibleReadableChannel.java:54)
2021-01-20T05:37:40.3594878Z 	at org.apache.maven.plugin.surefire.extensions.EventConsumerThread.read(EventConsumerThread.java:223)
2021-01-20T05:37:40.3598063Z 	at org.apache.maven.plugin.surefire.extensions.EventConsumerThread.decode(EventConsumerThread.java:150)
2021-01-20T05:37:40.3601135Z 	at org.apache.maven.plugin.surefire.extensions.EventConsumerThread.run(EventConsumerThread.java:113)
2021-01-20T05:37:40.3604480Z 
2021-01-20T05:37:40.3605642Z "fork-1-err-thread" #9054 daemon prio=5 os_prio=0 cpu=0.00ms elapsed=3516.28s tid=0x000001ca771e2800 nid=0x1d3c runnable  [0x00000029806fe000]
2021-01-20T05:37:40.3606817Z    java.lang.Thread.State: RUNNABLE
2021-01-20T05:37:40.3607965Z 	at java.io.FileInputStream.readBytes(java.base@11.0.9.1/Native Method)
2021-01-20T05:37:40.3611019Z 	at java.io.FileInputStream.read(java.base@11.0.9.1/FileInputStream.java:279)
2021-01-20T05:37:40.3612485Z 	at java.io.BufferedInputStream.fill(java.base@11.0.9.1/BufferedInputStream.java:252)
2021-01-20T05:37:40.3614089Z 	at java.io.BufferedInputStream.read1(java.base@11.0.9.1/BufferedInputStream.java:292)
2021-01-20T05:37:40.3617091Z 	at java.io.BufferedInputStream.read(java.base@11.0.9.1/BufferedInputStream.java:351)
2021-01-20T05:37:40.3618322Z 	- locked <0x00000000f6f1ffe0> (a java.io.BufferedInputStream)
2021-01-20T05:37:40.3619911Z 	at org.apache.maven.surefire.api.util.internal.Channels$3.readImpl(Channels.java:214)
2021-01-20T05:37:40.3624897Z 	at org.apache.maven.surefire.api.util.internal.AbstractNoninterruptibleReadableChannel.read(AbstractNoninterruptibleReadableChannel.java:54)
2021-01-20T05:37:40.3628541Z 	at sun.nio.cs.StreamDecoder.readBytes(java.base@11.0.9.1/StreamDecoder.java:274)
2021-01-20T05:37:40.3630036Z 	at sun.nio.cs.StreamDecoder.implRead(java.base@11.0.9.1/StreamDecoder.java:326)
2021-01-20T05:37:40.3631329Z 	at sun.nio.cs.StreamDecoder.read(java.base@11.0.9.1/StreamDecoder.java:178)
2021-01-20T05:37:40.3632487Z 	- locked <0x00000000f6f58a10> (a sun.nio.cs.StreamDecoder)
2021-01-20T05:37:40.3634861Z 	at java.io.Reader.read(java.base@11.0.9.1/Reader.java:189)
2021-01-20T05:37:40.3635775Z 	at java.util.Scanner.readInput(java.base@11.0.9.1/Scanner.java:882)
2021-01-20T05:37:40.3637056Z 	at java.util.Scanner.findWithinHorizon(java.base@11.0.9.1/Scanner.java:1796)
2021-01-20T05:37:40.3638306Z 	at java.util.Scanner.hasNextLine(java.base@11.0.9.1/Scanner.java:1610)
2021-01-20T05:37:40.3641862Z 	at org.apache.maven.surefire.extensions.util.LineConsumerThread.run(LineConsumerThread.java:67)
2021-01-20T05:37:40.3643669Z 
2021-01-20T05:37:40.3645411Z "VM Thread" os_prio=2 cpu=5218.75ms elapsed=10635.90s tid=0x000001ca47649000 nid=0x8c8 runnable  
2021-01-20T05:37:40.3646228Z 
2021-01-20T05:37:40.3647172Z "GC Thread#0" os_prio=2 cpu=70890.63ms elapsed=10635.92s tid=0x000001ca2ab3e800 nid=0x10d4 runnable  
2021-01-20T05:37:40.3647711Z 
2021-01-20T05:37:40.3648662Z "GC Thread#1" os_prio=2 cpu=71203.13ms elapsed=10632.78s tid=0x000001ca48388800 nid=0x960 runnable  
2021-01-20T05:37:40.3649357Z 
2021-01-20T05:37:40.3650253Z "G1 Main Marker" os_prio=2 cpu=687.50ms elapsed=10635.92s tid=0x000001ca2ab62800 nid=0x89c runnable  
2021-01-20T05:37:40.3652435Z 
2021-01-20T05:37:40.3653168Z "G1 Conc#0" os_prio=2 cpu=1079015.63ms elapsed=10635.92s tid=0x000001ca2ab63800 nid=0x1ad4 runnable  
2021-01-20T05:37:40.3653769Z 
2021-01-20T05:37:40.3654616Z "G1 Refine#0" os_prio=2 cpu=671.88ms elapsed=10635.92s tid=0x000001ca474b5800 nid=0x6b8 runnable  
2021-01-20T05:37:40.3655199Z 
2021-01-20T05:37:40.3656058Z "G1 Refine#1" os_prio=2 cpu=46.88ms elapsed=10620.96s tid=0x000001ca481f5800 nid=0x544 runnable  
2021-01-20T05:37:40.3658069Z 
2021-01-20T05:37:40.3659013Z "G1 Young RemSet Sampling" os_prio=2 cpu=390.63ms elapsed=10635.92s tid=0x000001ca2abee000 nid=0xbc4 runnable  
2021-01-20T05:37:40.3660213Z "VM Periodic Task Thread" os_prio=2 cpu=1390.63ms elapsed=10635.82s tid=0x000001ca48155800 nid=0x1080 waiting on condition  
2021-01-20T05:37:40.3660822Z 
2021-01-20T05:37:40.3661353Z JNI global refs: 20, weak refs: 0
2021-01-20T05:37:40.3661758Z 
2021-01-20T05:37:40.3662190Z Heap
2021-01-20T05:37:40.3662867Z  garbage-first heap   total 1405952K, used 1080599K [0x00000000a8000000, 0x0000000100000000)
2021-01-20T05:37:40.3663668Z   region size 1024K, 45 young (46080K), 1 survivors (1024K)
2021-01-20T05:37:40.3664446Z  Metaspace       used 335233K, capacity 342466K, committed 497416K, reserved 1437696K
2021-01-20T05:37:40.3665338Z   class space    used 37961K, capacity 40363K, committed 75648K, reserved 1015808K
2021-01-20T05:37:40.3667356Z 
2021-01-20T05:37:45.3991367Z ##[error]The operation was canceled.
```

See https://pipelines.actions.githubusercontent.com/5Na9LjmLgXLiLuzJ0Z1qHRbIDVUXUHNrKe6jPcqGdw5t1S8TiW/_apis/pipelines/1/runs/95388/signedlogcontent/59?urlExpires=2021-01-20T08%3A30%3A41.0458526Z&urlSigningMethod=HMACV1&urlSignature=8cBAZMXViTg9xZ5g4ZYP1fvCfOzwXzIs8m8OWuEBIOA%3D for the full content.

--- a/extensions/mongodb-client/deployment/src/test/java/io/quarkus/mongodb/MongoTestBase.java
+++ b/extensions/mongodb-client/deployment/src/test/java/io/quarkus/mongodb/MongoTestBase.java
@@ -41,6 +41,13 @@ public class MongoTestBase {
 
     @BeforeAll
     public static void startMongoDatabase() throws IOException {
+        try {
+            //JDK bug workaround
+            //https://github.com/quarkusio/quarkus/issues/14424
+            //force class init to prevent possible deadlock when done by mongo threads
+            Class.forName("sun.net.ext.ExtendedSocketOptions", true, ClassLoader.getSystemClassLoader());
+        } catch (ClassNotFoundException e) {
+        }
         String uri = getConfiguredConnectionString();
         // This switch allow testing against a running mongo database.
         if (uri == null) {

--- a/extensions/mongodb-client/deployment/src/test/java/io/quarkus/mongodb/MongoWithReplicasTestBase.java
+++ b/extensions/mongodb-client/deployment/src/test/java/io/quarkus/mongodb/MongoWithReplicasTestBase.java
@@ -175,6 +175,13 @@ public class MongoWithReplicasTestBase {
 
     private static IMongodConfig buildMongodConfiguration(String url, int port, final boolean configureReplicaSet)
             throws IOException {
+        try {
+            //JDK bug workaround
+            //https://github.com/quarkusio/quarkus/issues/14424
+            //force class init to prevent possible deadlock when done by mongo threads
+            Class.forName("sun.net.ext.ExtendedSocketOptions", true, ClassLoader.getSystemClassLoader());
+        } catch (ClassNotFoundException e) {
+        }
         final MongodConfigBuilder builder = new MongodConfigBuilder()
                 .version(Version.Main.V4_0)
                 .net(new Net(url, port, Network.localhostIsIPv6()));

--- a/extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClients.java
+++ b/extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClients.java
@@ -76,6 +76,14 @@ public class MongoClients {
     public MongoClients(MongodbConfig mongodbConfig, MongoClientSupport mongoClientSupport) {
         this.mongodbConfig = mongodbConfig;
         this.mongoClientSupport = mongoClientSupport;
+
+        try {
+            //JDK bug workaround
+            //https://github.com/quarkusio/quarkus/issues/14424
+            //force class init to prevent possible deadlock when done by mongo threads
+            Class.forName("sun.net.ext.ExtendedSocketOptions", true, ClassLoader.getSystemClassLoader());
+        } catch (ClassNotFoundException e) {
+        }
     }
 
     public MongoClient createMongoClient(String clientName) throws MongoException {

--- a/extensions/mongodb-client/runtime/src/test/java/io/quarkus/mongodb/reactive/MongoTestBase.java
+++ b/extensions/mongodb-client/runtime/src/test/java/io/quarkus/mongodb/reactive/MongoTestBase.java
@@ -54,6 +54,13 @@ public class MongoTestBase {
 
     @BeforeAll
     public static void startMongoDatabase() throws IOException {
+        try {
+            //JDK bug workaround
+            //https://github.com/quarkusio/quarkus/issues/14424
+            //force class init to prevent possible deadlock when done by mongo threads
+            Class.forName("sun.net.ext.ExtendedSocketOptions", true, ClassLoader.getSystemClassLoader());
+        } catch (ClassNotFoundException e) {
+        }
         String uri = getConfiguredConnectionString();
         // This switch allow testing against a running mongo database.
         if (uri == null) {

--- a/extensions/mongodb-client/runtime/src/test/java/io/quarkus/mongodb/reactive/MongoWithReplicasTestBase.java
+++ b/extensions/mongodb-client/runtime/src/test/java/io/quarkus/mongodb/reactive/MongoWithReplicasTestBase.java
@@ -183,6 +183,13 @@ public class MongoWithReplicasTestBase {
 
     private static IMongodConfig buildMongodConfiguration(String url, int port, final boolean configureReplicaSet)
             throws IOException {
+        try {
+            //JDK bug workaround
+            //https://github.com/quarkusio/quarkus/issues/14424
+            //force class init to prevent possible deadlock when done by mongo threads
+            Class.forName("sun.net.ext.ExtendedSocketOptions", true, ClassLoader.getSystemClassLoader());
+        } catch (ClassNotFoundException e) {
+        }
         final MongodConfigBuilder builder = new MongodConfigBuilder()
                 .version(Version.Main.V4_0)
                 .net(new Net(url, port, Network.localhostIsIPv6()));

--- a/integration-tests/mongodb-client/src/test/java/io/quarkus/it/mongodb/MongoTestResource.java
+++ b/integration-tests/mongodb-client/src/test/java/io/quarkus/it/mongodb/MongoTestResource.java
@@ -27,6 +27,13 @@ public class MongoTestResource implements QuarkusTestResourceLifecycleManager {
     @Override
     public Map<String, String> start() {
         try {
+            //JDK bug workaround
+            //https://github.com/quarkusio/quarkus/issues/14424
+            //force class init to prevent possible deadlock when done by mongo threads
+            Class.forName("sun.net.ext.ExtendedSocketOptions", true, ClassLoader.getSystemClassLoader());
+        } catch (ClassNotFoundException e) {
+        }
+        try {
             Version.Main version = Version.Main.V4_0;
             int port = 27018;
             LOGGER.infof("Starting Mongo %s on port %s", version, port);

--- a/integration-tests/mongodb-panache-kotlin/src/test/kotlin/io/quarkus/it/mongodb/panache/MongoTestResource.kt
+++ b/integration-tests/mongodb-panache-kotlin/src/test/kotlin/io/quarkus/it/mongodb/panache/MongoTestResource.kt
@@ -22,6 +22,13 @@ class MongoTestResource : QuarkusTestResourceLifecycleManager {
     private var mongod: MongodExecutable? = null
 
     override fun start(): Map<String, String> {
+        try {
+            //JDK bug workaround
+            //https://github.com/quarkusio/quarkus/issues/14424
+            //force class init to prevent possible deadlock when done by mongo threads
+            Class.forName("sun.net.ext.ExtendedSocketOptions", true, ClassLoader.getSystemClassLoader());
+        } catch (e: ClassNotFoundException) {
+        }
         return try {
             val version: Version.Main = Version.Main.V4_0
             val port = 27018

--- a/integration-tests/mongodb-panache/src/test/java/io/quarkus/it/mongodb/panache/MongoTestResource.java
+++ b/integration-tests/mongodb-panache/src/test/java/io/quarkus/it/mongodb/panache/MongoTestResource.java
@@ -26,6 +26,13 @@ public class MongoTestResource implements QuarkusTestResourceLifecycleManager {
     @Override
     public Map<String, String> start() {
         try {
+            //JDK bug workaround
+            //https://github.com/quarkusio/quarkus/issues/14424
+            //force class init to prevent possible deadlock when done by mongo threads
+            Class.forName("sun.net.ext.ExtendedSocketOptions", true, ClassLoader.getSystemClassLoader());
+        } catch (ClassNotFoundException e) {
+        }
+        try {
             Version.Main version = Version.Main.V4_0;
             int port = 27018;
             LOGGER.infof("Starting Mongo %s on port %s", version, port);

--- a/integration-tests/mongodb-rest-data-panache/src/test/java/io/quarkus/it/mongodb/rest/data/panache/MongoTestResource.java
+++ b/integration-tests/mongodb-rest-data-panache/src/test/java/io/quarkus/it/mongodb/rest/data/panache/MongoTestResource.java
@@ -22,6 +22,13 @@ public class MongoTestResource implements QuarkusTestResourceLifecycleManager {
     @Override
     public Map<String, String> start() {
         try {
+            //JDK bug workaround
+            //https://github.com/quarkusio/quarkus/issues/14424
+            //force class init to prevent possible deadlock when done by mongo threads
+            Class.forName("sun.net.ext.ExtendedSocketOptions", true, ClassLoader.getSystemClassLoader());
+        } catch (ClassNotFoundException e) {
+        }
+        try {
             Version.Main version = Version.Main.V4_0;
             int port = 27018;
             LOGGER.infof("Starting Mongo %s on port %s", version, port);

